### PR TITLE
VPA - Adding status object validator

### PR DESCRIPTION
**Pull request by krzysied**
Adding Validator interface that will be used by status readers.

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/utils/status/status_object.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ type Client struct {
 	holderIdentity       string
 }
 
+// Validator for the status object.
+type Validator interface {
+	IsStatusValid(leaseTimeout time.Duration) (bool, error)
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/utils/status/status_object.go]: can you change leaseTimeout to something more implementation neutral? 
It doesn't have to be a lease backing the status.

Comment (krzysied) [reply vertical-pod-autoscaler/pkg/utils/status/status_object.go]: Done. `statusTimeout` it is.

**Comment by bskiba** [comment]
:+1: 
/lgtm
/approve

**Pull request merged**